lua/grapple/ui/tags.lua

```diff
@@ -5,106 +5,175 @@ local popup = require("grapple.ui.popup")
 
 local M = {}
 
+---@class Grapple.Items
+---@field tags Grapple.TagTable
+---@field lines string[]
+---@field map_relpath_tag table<string, Grapple.Tag>
+
 ---@param scope Grapple.Scope
----@return { tags: Grapple.TagTable, lines: string[] }
+---@return Grapple.Items
 local function itemize(scope)
     local scoped_tags = _tags.tags(scope)
     local scope_path = _scope.resolve(scope)
     local sanitized_scope_path = string.gsub(scope_path, "%p", "%%%1")
 
     local lines = {}
+    local map_relpath_tag = {}
     for key, tag in pairs(scoped_tags) do
         local relative_path = string.gsub(tag.file_path, sanitized_scope_path .. "/", "")
         local text = " [" .. key .. "] " .. relative_path
         table.insert(lines, text)
+        map_relpath_tag[relative_path] = tag
     end
 
     return {
         tags = scoped_tags,
         lines = lines,
+        map_relpath_tag = map_relpath_tag,
     }
 end
 
+---@class Grapple.PartialTag
+---@field key string | integer
+---@field relative_path string
+
 ---@param line string
----@return Grapple.TagKey | nil
+---@return Grapple.PartialTag | nil
 local function parse(line)
-    local start, _end = string.find(line, "%[.*%]")
-    if not start or not _end then
-        log.warn("Unable to parse line into tag key. Line: " .. line)
+    if #line == 0 then -- no need to warn
         return nil
     end
-    local parsed_key = string.sub(line, start + 1, _end - 1)
-    return tonumber(parsed_key) or parsed_key
+    local pattern = "%[(.*)%] +(.*)"
+    for key, relative_path in string.gmatch(line, pattern) do
```

Comment:
Just because the gmatch actually split the matched groups, whereas match returns a string
```lua
print(string.match(" [0] grapple.nvim/lua/init.lua", "%[(.*)%] +(.*)"))
-- "0 grapple.nvim/lua/init.lua" as string, but can still split on " " ? might be better
```

Reply:
Nevermind ! I learnt that it works reading your refactor PR